cdap-master/src/main/java/co/cask/cdap/data/tools/AbstractUpgrade.java

```diff
@@ -0,0 +1,155 @@
+import com.google.gson.GsonBuilder;
+import com.google.inject.Injector;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Abstract class for Upgrade
+ */
+public abstract class AbstractUpgrade {
+
+  public static final String EMPTY_STRING = "";
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractUpgrade.class);
+  static final byte[] COLUMN = Bytes.toBytes("c");
+  static final String FORWARD_SLASH = "/";
+  static final String CDAP_WITH_FORWARD_SLASH = Constants.Logging.SYSTEM_NAME + FORWARD_SLASH;
+  static final String DEVELOPER_STRING = "developer";
+
+  static DatasetFramework namespacedFramework;
+  static DatasetFramework nonNamespaedFramework;
+  static TransactionExecutorFactory executorFactory;
+  static CConfiguration cConf;
+  static TransactionService txService;
+  static ZKClientService zkClientService;
+  static LocationFactory locationFactory;
+  static TransactionSystemClient txClient;
+  static DefaultStore defaultStore;
+  static final Gson GSON;
+
+  static {
+    GsonBuilder builder = new GsonBuilder();
+    ApplicationSpecificationAdapter.addTypeAdapters(builder);
+    GSON = builder.create();
+  }
+
+  /**
+   * Sets up a {@link DatasetFramework} instance for standalone usage.  NOTE: should NOT be used by applications!!!
+   */
+  public static DatasetFramework createRegisteredDatasetFramework(Injector injector)
+    throws DatasetManagementException, IOException {
+    CConfiguration cConf = injector.getInstance(CConfiguration.class);
+
+    DatasetDefinitionRegistryFactory registryFactory = injector.getInstance(DatasetDefinitionRegistryFactory.class);
+    DatasetFramework datasetFramework =
+      new NamespacedDatasetFramework(new InMemoryDatasetFramework(registryFactory),
+                                     new DefaultDatasetNamespace(cConf));
+    // TODO: this doesn't sound right. find out why its needed.
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "orderedTable"),
+                               new HBaseOrderedTableModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "metricsTable"),
+                               new HBaseMetricsTableModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "core"), new CoreDatasetsModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "fileSet"), new FileSetModule());
+
+    return datasetFramework;
+  }
+
+  /**
+   * Creates a non-namespaced {@link DatasetFramework} to access existing datasets which are not namespaced
+   */
+  public DatasetFramework createNonNamespaceDSFramework(Injector injector) throws DatasetManagementException {
```

Comment:
Lets discuss this in person. Its for the problem we saw where the table was created like cdap.system.cdap.system.table.name since the other framework is namespaced.